examples/todomvc/src/view.rs

```diff
@@ -0,0 +1,478 @@
+            if let Some(e) = wasm_bindgen::JsCast::dyn_ref::<web_sys::HtmlElement>(&parent) {
+                if e.dataset().get("id") != "" {
+                    res = Some(e.dataset().get("id"))
+                }
+            };
+            if None == res {
+                if let Some(ep) = parent.parent_node() {
+                    if let Some(dyn_el) = wasm_bindgen::JsCast::dyn_ref::<web_sys::HtmlElement>(&ep)
+                    {
+                        res = Some(dyn_el.dataset().get("id"));
+                    }
+                }
+            }
+            res.unwrap()
+        })
+    } else {
+        None
+    }
+}
+
+/// Presentation layer
+#[wasm_bindgen]
+pub struct View {
+    sched: RefCell<Rc<Scheduler>>,
+    todo_list: Element,
+    todo_item_counter: Element,
+    clear_completed: Element,
+    main: Element,
+    toggle_all: Element,
+    new_todo: Element,
+    callbacks: Vec<(web_sys::EventTarget, String, Closure<FnMut()>)>,
+}
+
+impl View {
+    /// Construct a new view
+    pub fn new(sched: Rc<Scheduler>) -> Option<View> {
+        let todo_list = Element::qs(".todo-list")?;
+        let todo_item_counter = Element::qs(".todo-count")?;
+        let clear_completed = Element::qs(".clear-completed")?;
+        let main = Element::qs(".main")?;
+        let toggle_all = Element::qs(".toggle-all")?;
+        let new_todo = Element::qs(".new-todo")?;
+        Some(View {
+            sched: RefCell::new(sched),
+            todo_list,
+            todo_item_counter,
+            clear_completed,
+            main,
+            toggle_all,
+            new_todo,
+            callbacks: Vec::new(),
+        })
+    }
+
+    pub fn init(&mut self) {
+        if let Some(window) = web_sys::window() {
+            if let Some(document) = window.document() {
+                let sched = self.sched.clone();
+                let set_page = Closure::wrap(Box::new(move || {
+                    dbg("new hash");
+                    if let Some(location) = document.location() {
+                        if let Ok(hash) = location.hash() {
+                            // TODO refactor back into fn
+                            // Was: &self.add_message(ControllerMessage::SetPage(hash));
+                            if let Ok(sched) = &(sched.try_borrow_mut()) {
+                                sched.add_message(Message::Controller(ControllerMessage::SetPage(
+                                    hash,
+                                )));
+                            }
+                            // TODO refactor back into fn
+                        }
+                    }
+                }) as Box<FnMut()>);
+
+                let window_et: web_sys::EventTarget = window.into();
+                window_et.add_event_listener_with_callback(
+                    "hashchange",
+                    set_page.as_ref().unchecked_ref(),
+                );
+                set_page.forget(); // Cycle collect this
```

Comment:
It looks like this is pretty common throughout the app (forgetting event listener callbacks). Is this happening often like with every click? If so, would it be good to figure out a way to deallocate these?